src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -238,10 +238,13 @@ protected override void Dispose(bool disposing)
         // NB: Other providers don't set Transaction
         public new virtual SqliteCommand CreateCommand() => new SqliteCommand { Connection = this, Transaction = Transaction };
         protected override DbCommand CreateDbCommand() => CreateCommand();
-        public new virtual SqliteTransaction BeginTransaction() => BeginTransaction(IsolationLevel.Unspecified);
-        protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel) => BeginTransaction(isolationLevel);
+        public new virtual SqliteTransaction BeginTransaction() => BeginTransactionImpl(IsolationLevel.Unspecified);
+        protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel) => BeginTransactionImpl(isolationLevel);
```

Comment:
Keep these as `BeginTransaction` so wrapping providers will have to override `BeginTransaction(IsolationLevel)`.